sympy/functions/elementary/integers.py

```diff
@@ -132,9 +132,12 @@ def _eval_number(cls, arg):
             return arg.approximation_interval(Integer)[0]
 
     def _eval_nseries(self, x, n, logx, cdir=0):
```

Comment:
I think simply adding a check for `zoo` in `_eval_nseries` should be sufficient instead of adding `_eval_aseries`, as anyways `_eval_nseries` is being called first.

Reply:
This is expected to return a power series but that does not exist at oo or -oo.

Reply:
So should `_eval_nseries` check for infinite values as is being done in- 

https://github.com/sympy/sympy/blob/9d610300d8b9b38c66b0e26be642fe7a3e15ef1e/sympy/core/function.py#L696-L697

and if the value is infinite, it should call `_eval_aseries`? Because `_eval_aseries` would just be a one-line function which returns `arg` from `floor(arg)` (in fact, `arg` is already found out in `_eval_nseries`, and we can just pass that, and the function would just be returning its own argument). Is that better than this current implementation or am I missing something?

Reply:
Ideally, nseries should return some indication that there is no power series, and then the caller would try to find an asymptotic expansion. One possibility would be to return None but it seems that currently the original expression itself is often returned.

The asymptotic expansion could probably be `floor(1/x) = 1/x + O(1)`.

Reply:
Adding `O(1)` is causing `Sum(S.Half**floor(n/2), (n, 1, oo)).is_convergent()` to not terminate. The `Sum` being calculated with and without `O(1)` is symbolically the same, but `is_convergent()` method does not terminate, so I have removed `O(1)`.

Reply:
The order term should usually include symbol and location. O(1) can have many meanings.

Reply:
I have tried with `Order(1)`, `Order(1, x)` and `Order(1, (x, 0))`, all of them were non-terminating. Why is it better to return `O(1)` with the arg?

Reply:
How about something like `O(1, (x, oo))`? Returning the argument alone is not correct. The caller can remove the order term if necessary.

Reply:
In that case, `is_convergent` is passing but the original issue reverts back to output 0.